src/main/java/uk/gov/companieshouse/web/companydocuments/util/EnumerationConstantsMapper.java

```diff
@@ -0,0 +1,32 @@
+package uk.gov.companieshouse.web.companydocuments.util;
+
+import java.util.Map;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+@Component
+public class EnumerationConstantsMapper {
+
+    private Map<String, String> companyTypes;
+
+    private Map<String, String> companyStatuses;
+
+    @Autowired
+    public EnumerationConstantsMapper(YamlResourceMapper yamlResourceMapper) {
+
+        Map<String, Object> enumerationConstants = yamlResourceMapper.fetchYamlFromFile("api-enumerations/constants.yml");
+
+        this.companyTypes = (Map<String, String>) enumerationConstants.get("company_type");
+        this.companyStatuses = (Map<String, String>) enumerationConstants.get("company_status");
+    }
+
+    public String getCompanyType(String enumeration) {
```

Comment:
Should this complain (throw ex) if enumeration does not exist in companyTypes?

Reply:
You're quite right. I was hoping to avoid it as it makes the code a little messy - especially given that we should never encounter the scenario. That said, a safety guard isn't a bad shout at all. Added in [5696f8a](https://github.com/companieshouse/company-documents.web.ch.gov.uk/pull/10/commits/5696f8a83b72abac4f4b2aa5b454df5b70aeea50)